src/query/storages/iceberg/src/table.rs

```diff
@@ -405,4 +418,56 @@ impl Table for IcebergTable {
     fn has_exact_total_row_count(&self) -> bool {
         true
     }
+
+    #[async_backtrace::framed]
+    async fn navigate_to(
+        &self,
+        navigation: &TimeNavigation,
+        _abort_checker: AbortChecker,
+    ) -> Result<Arc<dyn Table>> {
+        let snapshot_id = match navigation {
+            TimeNavigation::TimeTravel(np) => match np {
+                NavigationPoint::SnapshotID(sid) => {
+                    let sid = sid.parse::<i64>()?;
+                    if self.table.metadata().snapshot_by_id(sid).is_some() {
+                        Some(sid)
+                    } else {
+                        None
+                    }
+                }
+                NavigationPoint::TimePoint(dt) => {
+                    let ts = dt.timestamp_millis();
+                    self.table
+                        .metadata()
+                        .history()
+                        .iter()
+                        .filter(|log| log.timestamp_ms < ts)
+                        .max_by_key(|log| log.timestamp_ms)
+                        .map(|s| s.snapshot_id)
+                }
+                _ => {
+                    return Err(ErrorCode::Unimplemented(format!(
+                            "Time travel operation is not supported for the table '{}', which uses the '{}' engine.",
+                            self.name(),
+                            self.get_table_info().engine(),
+                        )));
+                }
+            },
+            _ => {
+                return Err(ErrorCode::Unimplemented(format!(
+                    "Time travel operation is not supported for the table '{}', which uses the '{}' engine.",
+                    self.name(),
+                    self.get_table_info().engine(),
+                )));
+            }
+        };
+
+        let (table, statistics) = Self::parse_engine_options(&self.info.meta.engine_options)?;
+        Ok(Arc::new(IcebergTable {
```

Comment:
better to throw an error if no snapshot found, which keep consistent with fuse table

Reply:
Done